plugin/src/main/java/net/codingarea/challenges/plugin/utils/bukkit/nms/NMSProvider.java

```
@@ -0,0 +1,116 @@
+
+import net.codingarea.challenges.plugin.Challenges;
+import net.codingarea.challenges.plugin.utils.bukkit.nms.implementations.v1_13.*;
+import net.codingarea.challenges.plugin.utils.bukkit.nms.implementations.v1_17.BorderPacketFactory_1_17;
+import net.codingarea.challenges.plugin.utils.bukkit.nms.implementations.v1_17.CraftPlayer_1_17;
+import net.codingarea.challenges.plugin.utils.bukkit.nms.implementations.v1_17.PacketBorder_1_17;
+import net.codingarea.challenges.plugin.utils.bukkit.nms.implementations.v1_18.PacketBorder_1_18;
+import net.codingarea.challenges.plugin.utils.bukkit.nms.implementations.v1_18.PlayerConnection_1_18;
+import net.codingarea.challenges.plugin.utils.bukkit.nms.type.*;
+import org.bukkit.World;
+import org.bukkit.entity.Player;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * @author sehrschlechtYT | https://github.com/sehrschlechtYT
+ * @since 2.2.3
+ */
+public class NMSProvider {
+    private final static int majorVersion;
+    private static BorderPacketFactory borderPacketFactory;
+
+    static {
+        majorVersion = ReflectionUtil.getMajorVersion();
+        if(versionIsAtLeast(17)) {
+            borderPacketFactory = new BorderPacketFactory_1_17();
+        } else if(versionIsAtLeast(13)) {
+            borderPacketFactory = new BorderPacketFactory_1_13();
+        }
+    }
+
+
+    /**
+     * Creates a world server for the given world
+     * @param world The world
+     * @return The world server, or null if the server version is not supported
+     */
+    @Nullable
+    public static WorldServer createWorldServer(World world) {
+        try {
+            if(versionIsAtLeast(13)) {
+                return new WorldServer_1_13(world);
+            }
+        } catch (ClassNotFoundException exception) {
+            Challenges.getInstance().getLogger().error("Failed to create WorldServer instance for version {}:", majorVersion, exception);
+        }
+        return null;
```

Should throw exception instead of marking as nullable.